src/process_stash.hpp

```diff
@@ -0,0 +1,89 @@
+#ifndef IRODS_HTTP_PROCESS_STASH_HPP
+#define IRODS_HTTP_PROCESS_STASH_HPP
+
+/// \file
+
+// Boost.Any is used instead of the implementation provided by the standard library
+// because it produces the correct results when used across shared library boundaries.
+#include <boost/any.hpp>
+
+#include <functional>
+#include <optional>
+#include <string>
+#include <vector>
+
+/// Defines the set of free functions used to manage the process stash.
+///
+/// In iRODS, a process stash is a key-value store which allows a process to store arbitrary
+/// data in local memory. Insertions of data generate unique handles which allow retrieval of
+/// the data. No two processes share the same data (unless one is a child process).
+///
+/// \since 4.2.12
+namespace irods::http::process_stash
+{
+    /// Inserts an object into the process stash.
+    ///
+    /// This function is thread-safe.
+    ///
+    /// \param[in] _value The object to insert.
+    ///
+    /// \returns A string representing the handle to the inserted object.
+    ///
+    /// \since 4.2.12
+    auto insert(boost::any _value) -> std::string;
+
+    /// Searches the process stash for the value associated with a specific key.
+    ///
+    /// This function is thread-safe.
+    ///
+    /// \param[in] _key The string which maps to the value of interest.
+    ///
+    /// \returns A pointer to the value of interest.
+    /// \retval A boost::any object if the key exists.
+    /// \retval A boost::any object containing a nullptr otherwise.
+    ///
+    /// \since 4.2.12
```

Comment:
Just making a note here vis-a-vis porting this back to irods/irods...

Do we need to follow our deprecation model for this (or any other function) since we are changing the interface and this is client-facing? I realize it's a relatively new, C++ client library, so maybe we won't need to, but just wanted to pose the question.

Should we follow the deprecation model, I guess we would need a differently named function at first (since you can't overload based on return type) and then do a monkey-head later to get back to the desired function name.

Reply:
I don't think we need to follow a deprecation model for this. The interfaces are small enough to eyeball and spot differences.

It's also possible that this be merged into irods/irods as an alternative to the existing process stash.

Let's let it ride for a while and see what it becomes.

Reply:
Resolving.